backend/LexBoxApi/GraphQL/UserMutations.cs

```diff
@@ -13,56 +16,75 @@ public class UserMutations
 {
     [Error<NotFoundException>]
     [Error<DbError>]
+    [Error<InvalidFormatException>]
     [ProducesResponseType(StatusCodes.Status401Unauthorized)]
     [UseMutationConvention]
-    public async Task<User> ChangeUserAccountData(
+    public Task<User> ChangeUserAccountData(
         LoggedInContext loggedInContext,
         ChangeUserAccountDataInput input,
-        LexBoxDbContext dbContext)
+        LexBoxDbContext dbContext,
+        EmailService emailService,
+        LexAuthService lexAuthService
+    )
     {
         if (loggedInContext.User.Id != input.UserId) throw new UnauthorizedAccessException();
-        var user = await dbContext.Users.FindAsync(input.UserId);
-        if (user is null) throw new NotFoundException("User not found");
-        // below works to change email
-        // minimum email = a@a.a
-        // if (input.Email is not null && input.Email != ""){
-        //     if (input.Email.Contains("@") == false || input.Email.Length < 3){
-        //         throw new RequiredException("Email does not match requirements");
-        //     }
-        //     user.Email = input.Email;
-        // }
-
-        if (!String.IsNullOrEmpty(input.Name))
-        {
-            user.Name = input.Name;
-        }
-
-        await dbContext.SaveChangesAsync();
-        return user;
+        return UpdateUser(input.UserId, input.Name, input.Email, dbContext, emailService, lexAuthService);
     }
 
     [Error<NotFoundException>]
     [Error<DbError>]
+    [Error<InvalidFormatException>]
     [UseMutationConvention]
     [AdminRequired]
-    public async Task<User> ChangeUserAccountByAdmin(ChangeUserAccountByAdminInput input, LexBoxDbContext dbContext)
+    public Task<User> ChangeUserAccountByAdmin(
+        ChangeUserAccountByAdminInput input,
+        LexBoxDbContext dbContext,
+        EmailService emailService,
+        LexAuthService lexAuthService
+    )
     {
-        var user = await dbContext.Users.FindAsync(input.UserId);
+        return UpdateUser(input.UserId, input.Name, input.Email, dbContext, emailService, lexAuthService);
+    }
+
+    private static async Task<User> UpdateUser(
```

Comment:
can we either use inheritance or just a single type for the `ChangeUserInput` and use that here instead of a bunch of parameters.